packages/runner/src/run.ts

```diff
@@ -561,7 +563,7 @@ export async function startTests(specs: string[] | FileSpecification[], runner:
 
   await runner.onAfterRunFiles?.(files)
 
-  await sendTasksUpdate(runner)
+  sendTasksUpdate(runner)
```

Comment:
Seems dangerous that it's not awaited. We awaited here specifically so there is no situation where `onTaskUpdate` is sent after another test started running

Reply:
Since `sendTasksUpdate` is (looks) synchronous, I'm not sure what needs awaiting. The approach is changed from "debounce" to "throttle", so `sendTasksUpdate` only serves as flushing if anything remained.

Reply:
Yeah, that's the point. We need to flush anything that's remaining

Reply:
senTaskUpdate is also not sync, it returns a promise

Reply:
It's hard to get around my head, so I changed an approach a bit for await promises. https://github.com/vitest-dev/vitest/pull/7700/commits/44ff2a0428b23ba623db491303c75abdc8c8782c